### PR TITLE
refactor: convert panel actions to enums

### PR DESCRIPTION
**Pull request by midgleyc**

### CODE REVIEW

File: src/net/sourceforge/kolmafia/swingui/panel/ItemManagePanel.java [hunk 1]
```diff
@@ -370,7 +372,7 @@ protected int getDesiredItemAmount(
       final String itemName,
       final int itemCount,
       final String message,
-      final int quantityType) {
+      final QuantityType quantityType) {
     int quantity = 0;
     switch (quantityType) {
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/swingui/panel/ItemManagePanel.java:377]: Probable out of scope for this simple change, but making this an enum might allow this to be a switch expression.

There are other issues - break at the end of new-style switch block, instanceof followed by casts, etc. - but those sre not related to enumitude.

**Review by Veracity0 — APPROVED** [review]

**Comment by midgleyc** [comment]
may as well while we're here.

I think all the switches are already switch expressions?

**Comment by Veracity0** [comment]
> I think all the switches are already switch expressions?

Not quite. The method you just fixed is like this:
```
    method {
      int quantity = 0;
      switch (quantityType)
      ... case statements to calculate/set quantity
      }
      return quantity;
    }
```

**Comment by midgleyc** [comment]
I've converted more switch expressions, but I've left that one. It is a switch expression (`case ->` instead of `case:`), just the expression isn't used. It's a nontrivial conversion as `quantity` changes in the middle of some expressions, and there aren't tests around there.

**Comment by Veracity0** [comment]
> I've converted more switch expressions, but I've left that one. It is a switch expression (`case ->` instead of `case:`), just the expression isn't used. It's a nontrivial conversion as `quantity` changes in the middle of some expressions, and there aren't tests around there.

Fair enough.

**Review by Veracity0 — APPROVED** [review]

**Pull request merged**